pkg/util/test/comparisons.go

```diff
@@ -0,0 +1,45 @@
+package test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	core "k8s.io/client-go/testing"
+)
+
+// CompareActions checks slices of actual and expected Actions
+// for equality. It checks that the lengths of the slices are
+// the same, that each actual Action has a corresponding expected
+// Action, and that each expected Action has a corresponding
+// actual Action.
+func CompareActions(t *testing.T, expected, actual []core.Action) {
+	assert.Len(t, actual, len(expected))
+
+	for _, e := range expected {
+		found := false
+		for _, a := range actual {
+			if reflect.DeepEqual(e, a) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("missing expected action %#v", e)
```

Comment:
One thing that's annoying about `%#v` for `PatchAction`s is that it prints out the patch bytes in hex. We don't need to do this now, but it might be nice to have a custom printer for `PatchAction`s.